### ISSUE TITLE
What's wrong with this code

### ISSUE THREAD

**Issue by sassanh** [jazzband/django-oauth-toolkit#567]
Based on the logic of this repo's maintainer https://github.com/evonove/django-oauth-toolkit/pull/456#issuecomment-285223889 this method https://github.com/evonove/django-oauth-toolkit/blob/master/oauth2_provider/oauth2_backends.py#L176 is "most definitly not right" as it's "passing around a list of tuples in one instance, an empty string in another". Please fix it.

**Comment by jdelic** [comment]
@jleclanche I was going through issues on this repo to see where I could help. I looked at the original PR #456 for this issue which you closed without comment on April 14th. 

Can you perhaps elaborate on why catching `AttributeError` in addition to `ValueError` in [oauth2_provider.oauth2_backends](https://github.com/jazzband/django-oauth-toolkit/blob/86e8f9b22c9d5957b8ff6097208de69758a3c013/oauth2_provider/oauth2_backends.py#L182) is not a straight-forward bugfix that should be made?

**Comment by sassanh** [comment]
While a pull request has an open discussion going on waiting for the repo owner to answer, closing it by the repo owner instead of clarifying why he's against the PR is "most definitely" against the spirit of open source. I'm glad this repo is migrated to jazzband as there were no room for progress before this migration in my opinion.

The issue here is clear: there was a piece of code in this repo that based on @jleclanche logic wasn't correct (it returned a list of tuples in one instance and an empty string in another) it was there before that PR and after it. That PR didn't introduce this problem it just solved another problem: sending bulk creation/update requests to drf were not possible when django-oauth-toolkit were in use and that PR solved this problem. @jleclanche disagreed with the PR because of the problem already existed in his repo. After I clarified the problem wasn't introduced by my PR and was already there he didn't add any comment, 2 other people other than me came and agreed with me and mentioned him to hear his opinion and he didn't add anything and instead he closed the PR. That's just not how open source works.
After migrating repo to jazzband I could simply merge my PR but I decided to wait for the opinion of the community. I'm waiting for a conclusion but if no one has any clarification why this PR is not a good idea then I'm going to merge it myself as already 3 out of 3 people reviewed it and found no problem with it and 2 of them thought it's good (https://github.com/jazzband/django-oauth-toolkit/pull/456#issuecomment-300817197, https://github.com/jazzband/django-oauth-toolkit/pull/456#discussion_r123510123) and @jdelic seems to be positive with it too.
